pkg/controller/v1alpha2/trial/util/status_util.go

```diff
@@ -96,5 +99,58 @@ func UpdateTrialStatusCondition(instance *trialsv1alpha2.Trial, deployedJob *uns
 func UpdateTrialStatusObservation(instance *trialsv1alpha2.Trial, deployedJob *unstructured.Unstructured) error {
 
 	// read GetObservationLog call and update observation field
+	objectiveMetricName := instance.Spec.Objective.ObjectiveMetricName
+	request := &api_pb.GetObservationLogRequest{
+		TrialName:  instance.Name,
+		MetricName: objectiveMetricName,
+	}
+	if reply, err := common.GetObservationLog(request); err != nil {
+		return err
+	} else {
+		if reply.ObservationLog != nil {
+			bestObjectiveValue := getBestObjectiveMetricValue(reply.ObservationLog.MetricLogs, instance.Spec.Objective.Type)
+			if bestObjectiveValue != nil {
+				if instance.Status.Observation == nil {
+					instance.Status.Observation = &commonv1alpha2.Observation{}
+					metric := commonv1alpha2.Metric{Name: objectiveMetricName, Value: *bestObjectiveValue}
+					instance.Status.Observation.Metrics = []commonv1alpha2.Metric{metric}
+				} else {
+					for index, metric := range instance.Status.Observation.Metrics {
+						if metric.Name == objectiveMetricName {
+							instance.Status.Observation.Metrics[index].Value = *bestObjectiveValue
+						}
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func getBestObjectiveMetricValue(metricLogs []*api_pb.MetricLog, objectiveType commonv1alpha2.ObjectiveType) *float64 {
+	metricLogSize := len(metricLogs)
+	if metricLogSize == 0 {
+		return nil
+	} else if metricLogSize == 1 {
```

Comment:
Is this case needed? I think the else block would take care of it.

Reply:
done